src/custom-element.js

```diff
@@ -106,15 +106,19 @@ export function upgradeOrRegisterElement(win, name, toClass) {
     //    implementation.
     const element = stub.element;
     if (element.tagName.toLowerCase() == name) {
-      try {
-        element.upgrade(toClass);
-      } catch (e) {
-        reportError(e, this);
-      }
+      tryUpgradeElement_(element, toClass);
     }
   }
 }
 
+/** @private */
+function tryUpgradeElement_(element, toClass) {
```

Comment:
If we want to do this we need to make a coding convention for not inline, such as checking for "NoInline" here https://github.com/ampproject/amphtml/blob/master/build-system/runner/src/org/ampproject/AmpCodingConvention.java#L61